Rubberduck.VBEditor.VBA/SafeComWrappers/VB/VBProject.cs

```diff
@@ -100,7 +100,17 @@ public override int GetHashCode()
 
         public IReadOnlyList<string> ComponentNames()
         {
-            return VBComponents.Select(component => component.Name).ToArray();
+            var names = new List<string>();
+            using (var components = VBComponents)
+            {
+                foreach(var component in components)
```

Comment:
I generally prefer stacked `using`'s, but a stacked `foreach` is as dangerous as `if (foo) bar;`. I probably would have done something like:

`return VBComponents.Select(component => { var name = component.Name; component.Dispose(); return name; }).ToArray();`

Reply:
FWIW, I'm told to avoid doing LINQ in VBEditor projects.